src/Adapter/MSTest.TestAdapter/Helpers/MethodRunner.cs

```diff
@@ -83,6 +91,91 @@ internal static class MethodRunner
                 throw realException;
             }
 
+            throw;
+        }
+    }
+
+#if NET6_0_OR_GREATER
+    [System.Runtime.Versioning.SupportedOSPlatform("windows")]
+#endif
+    private static TestFailedException? RunWithTimeoutAndCancellationWithCustomThread(
```

Comment:
consider: add STA to the name

Reply:
yes please, `RunWithTimeoutAndCancellationWithSTAThread` is definitely a better name. **Custom** suggests that we are not even using the `System.Threading.Thread`.